### DISCUSSION TITLE
Compose and probability

### DISCUSSION THREAD

**Discussion by javier-alvarez** [Project-MONAI/MONAI#2387] [Q&A]
It looks like Compose does not respect the probability parameter on the random transforms.

Any ideas if this is by design or a bug?

**Comment by Nic-Ma** [comment]
Hi @javier-alvarez ,

Thanks for your interest here.
Could you please help share more details about your question or the bug you found?
I don't quite understand.

Thanks.

**Reply by javier-alvarez** [reply]
transform = RandomAfine(prob=0.5) 
transform(data) -> works as expected and half the time applies the transform
transform_compose = Compose([RandomAfine(prob=0.5)])
transform_compose(data) -> gets applied all the time

**Reply by Nic-Ma** [reply]
Hi @javier-alvarez ,

OK, thanks for raising this issue, let me try to reproduce locally and fix it.

**Reply by wyli** [reply]
I think it is randomised as expected, for example:
```py
import numpy as np
import torch

from monai.transforms import Compose, RandAffine

affine = RandAffine(prob=0.5, rotate_range=[np.pi / 4], padding_mode="zeros")
c_aff = Compose([affine])
c_aff.set_random_state(41)  # deterministic

runs = []
for _ in range(10):
    output = c_aff(torch.ones((1, 3, 3, 3)))
    print(c_aff.transforms[0]._do_transform)
    runs.append(output)

print(runs)
```

the  output `_do_transform` flags are
```
False
True
True
True
False
True
True
False
True
True
```

**Reply by javier-alvarez** [reply]
I was calling this on the python console multiple times: trans({"image":np.array([[[0,255,0,255]]]), "labels":np.array([[[0,1,0,1]]])})

and it always returns the same thing. However, in the loop it works. Any ideas?

many thanks for your quick help
